plugins/circleci/src/circleci-config.ts

```diff
@@ -109,7 +109,16 @@ export const generateConfigWithJob = (options: JobGeneratorOptions): CircleCISta
         ),
         ...matrixBoilerplate()
       }
-    : { requires: options.requires, ...nightlyBoilerplate }
+    : {
+        // only require the latest Node version of a matrix job in order to
+        // avoid workspace conflicts
+        requires: options.requires.map((dep) => `${dep}-node`),
+        // append the default executor name to the job name so that multiple
+        // non-matrix jobs can be chained one after another without having to
+        // know whether a matrix job precedes them or not
+        name: `${options.name}-node`,
+        ...nightlyBoilerplate
+      }
```

Comment:
**question(non-blocking):** do we choose which version of node is the primary? When we start supporting Node 20, we will want the follow jobs to require the Node 18 build and test jobs.

Reply:
yes, whichever is the first version in the `nodeVersion` option will be the default `node` executor. I've tried to make that clear in the [docs](https://github.com/Financial-Times/dotcom-tool-kit/blame/b60dad0df89902ea127a90dcddf036980cf5808e/plugins/circleci/readme.md#L32) too